src/dune_rules/merlin.ml

```diff
@@ -414,7 +435,7 @@ module Unprocessed = struct
         Path.Set.union src_dirs
           (Path.Set.of_list_map ~f:Path.source more_src_dirs)
       in
-      { Processed.stdlib_dir; src_dirs; obj_dirs; flags; extensions }
+      { Processed.stdlib_dir; src_dirs; obj_dirs; flags; extensions; mode }
```

Comment:
Processed shouldn't include the mode. The flags should already contain what's needed.

Reply:
Sorry, I spent some time trying to implement this, but I don't understand how it is possible.

`flags` includes only the compiler flags, and it's dumped in a single `FLG` directive here:

https://github.com/ocaml/dune/blob/75d5b4083857bee66f69b217aa6ca94147c2475a/src/dune_rules/merlin.ml#L92-L95

We want melange `FLG` to be a separate directive. I can replace `mode` field in processed with something like `melange_ppx_flag` of type, but something else needs to be passed to processed in order to add this new directive.

Reply:
Why does it need to be a separate `FLG` directive? We manage to put all the other flags in one directive.

Reply:
@rgrinberg Any thoughts on the above? Considering there were already two `FLG` directives, should `melc` ppx be made part of any of them or could it exist on its own? If anything, it should  be part of the second, but I don't think it's possible at the moment as it's not a user defined ppx (so can't be compiled with ppx driver).

Reply:
This is hacky but it looks good enough for now. A few points:

1. We shouldn't hard code the path of `melc`. We should use the absolute path that was resolved by dune
2. We should make sure that `melc` is actually built if it's defined in the same workspace.

In any case, if @anmonteiro is fine with this temporary hack then we can merge it.

Reply:
I agree we should fix 1. & 2. before merging.

What's the temporary hack? the 2 `FLG` directives?

Reply:
Yes. I would expect all these melange ppx's to be ported to ppxlib and enabled explicitly.

Reply:
that's the goal for the ReactJS PPX, as highlighted in https://github.com/ocaml/dune/pull/6476#discussion_r1024322957.

I don't yet have a clear plan for the builtin PPX, which currently is very tightly integrated into the compiler.